creusot-contracts/src/logic/fmap.rs

```diff
@@ -0,0 +1,124 @@
+use crate::{logic::Mapping, util::*, *};
+
+pub struct FMap<K, V: ?Sized>(pub Mapping<K, Option<SizedW<V>>>);
```

Comment:
As written, I think this is unsound, at the very least we should need the type to be opaque since otherwise it seems like there is a bug lurking akin to the `&[T]` and `Seq<T>` bug I discovered last week?